packages/client/wallets/smart-wallet/src/blockchain/wallets/passkey.ts

```diff
@@ -102,6 +106,32 @@ export class PasskeyAccountService {
             Authorization: `Bearer ${user.jwt}`,
         };
     }
+
+    private mapError(error: any, passkeyName: string) {
+        if (error.code === "ERROR_PASSTHROUGH_SEE_CAUSE_PROPERTY" && error.name === "NotAllowedError") {
+            return new PasskeyPromptError(passkeyName);
+        }
+
+        return error;
+    }
+
+    private decorate<Client extends KernelSmartAccount<EntryPoint>>(account: Client, passkeyName: string): Client {
+        return new Proxy(account, {
+            get: (target, prop, receiver) => {
+                if (prop !== "signUserOperation") {
+                    return Reflect.get(target, prop, receiver);
+                }
+
+                return async (...args: Parameters<Client["signUserOperation"]>) => {
+                    try {
+                        return await Reflect.get(target, prop, receiver).call(target, ...args);
+                    } catch (error) {
+                        throw this.mapError(error, passkeyName);
+                    }
+                };
+            },
+        });
+    }
```

Comment:
making sure I understand correctly. Here we're wrapping the viem client that we'd normally pass back such that when they call `signUserOperation`, we catch errors and map them to a friendlier error via `mapError`?
Nice, just the types scare me lol

Reply:
That generic is named poorly, we're actually wrapping the viem/permissionless `account`, any of the txn method we expose from the our viem client on `EVMSmartWallet`, `sendTransaction`, `sendUserOperation`, or `writeContract`, call this `signUserOperation` method under the hood, so by mapping errors here we get this error handling everywhere.

Reply:
Writing this out though I need to make sure that message signing is covered too

Reply:
Covered all signing method, and improved types where it matters!